Add OnlineData source and L1T quick collection workpace

abrinke1 opened the pull request:
Hi all,

These modifications should in principle be "transparent" to any previous uses of the AutoDQM tool.  I've added the capability to access the OnlineData, i.e. histograms used in the Online DQM.

I also added configs corresponding to histograms in the L1T workspace seen by shifters, as listed here:
https://github.com/dmwm/deployment/blob/master/dqmgui/layouts/l1t-layouts.py

Please take a look, and try it out, and let me know if there's anything you think should be changed!

Best regards,
Andrew

chadfreer commented on autodqm/dqm.py:
```diff
@@ -23,6 +23,19 @@
 StreamProg = namedtuple('StreamProg', ('cur', 'total', 'path'))
 DQMRow = namedtuple('DQMRow', ('name', 'full_name', 'url', 'size', 'date'))
 
+## Map from subsystem to Online DQM directory
```
Why is this OnlineMap required? As far as I understand, it reads this map to find the proper directory to grab the ROOT file. Perhaps we add a mapping file to the config directory instead of putting it in the dqm.py. This will be more obvious for future PRs

chadfreer replied:
I think a simple mapping in the configs is best. This way PRs in the future adding subsystems do not need to modify any python code.

abrinke1 replied:
Hi @chadfreer, I've added the mapping here:
https://github.com/AutoDQM/AutoDQM/pull/70/commits/d5d177d3a2addc8564abeeb9b51ce25652807d96

Let me know if you see any issues.

chadfreer commented on autodqm/dqm.py:
```diff
@@ -41,78 +54,135 @@ def __init__(self, cert, db, cache=None, workers=16):
 
-        # Get all run directories for this sample
-        macrorun_rows = _resolve(self._fetch_dqm_rows(sample_url)).data
+        isOnline = (series.startswith('000') and series.endswith('xxxx'))
```
This function is passed the dqmsource. Why do we need this isOnline variable?

abrinke1 replied:
Indeed, previously this "isOnline" switch led to different behavior, but in subsequent development there came to be no difference in the fetch_sample_list function.  I will delete this.

chadfreer replied:
Ok, great. Once removed this section is fine.

chadfreer commented on config/L1T_workspace_slim.json:
```diff
@@ -0,0 +1,151 @@
+{
```
Why are we adding a slim version? Is this a shifter specific list?

chadfreer replied:
Ah I see. If you think it is useful for shifters, I think having it in is useful in order to get feedback.

abrinke1 commented:
Hi @chadfreer , I think I've addressed all your comments.  Others are still welcome to take a look, and test it out.  @chosila found a couple of bugs in testing which I fixed here:
https://github.com/AutoDQM/AutoDQM/pull/70/commits/e0010abd155071f60c76b6ac46fd385c1c338f72
https://github.com/AutoDQM/AutoDQM/pull/70/commits/4f4e20ad470f85fcbab01ae4ecd9e80aab22b466
